src/euler/staking/StakeableEulerERC4626Factory.sol

```diff
@@ -0,0 +1,114 @@
+        rewardsDistribution = rewardsDistribution_;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Staking functions
+    /// -----------------------------------------------------------------------
+
+    /// @notice Returns how much was earned during staking
+    function reward(StakeableEulerERC4626 vault) public view returns (address rewardsToken, uint256 earned) {
+        return vault.reward();
+    }
+
+    /// @notice Allows owner to set or update a new staking contract. Will claim rewards from previous staking if available
+    function updateStakingAddress(StakeableEulerERC4626 vault, uint256 rewardIndex, address recipient) external onlyOwner {
+        vault.updateStakingAddress(rewardIndex, recipient);
+    }
+
+     /// @notice Allows owner to claim rewards and stop staking all together
+    function stopStaking(StakeableEulerERC4626 vault, address recipient) external onlyOwner {
+        vault.stopStaking(recipient);
+    }
+
+    /// @notice Allows owner to stake a certain amount of tokens
+    function stake(StakeableEulerERC4626 vault, uint256 amount) external onlyOwner {
+        vault.stake(amount);
+    }
+
+    /// @notice Allows owner to unstake a certain amount of tokens
+    function unstake(StakeableEulerERC4626 vault, uint256 amount) external onlyOwner {
+        vault.unstake(amount);
+    }
+
+    /// @notice Allows owner to claim all staking rewards
+    function claimReward(StakeableEulerERC4626 vault, address recipient) public onlyOwner returns (address rewardsToken, uint256 earned) {
+        return vault.claimReward(recipient);
+    }
+
+    /// -----------------------------------------------------------------------
+    /// External functions
+    /// -----------------------------------------------------------------------
+
+    /// @inheritdoc ERC4626Factory
+    function createERC4626(ERC20 asset) external virtual override returns (ERC4626 vault) {
+        address eTokenAddress = markets.underlyingToEToken(address(asset));
+        if (eTokenAddress == address(0)) {
+            revert StakeableEulerERC4626Factory__ETokenNonexistent();
+        }
+
+        vault = new StakeableEulerERC4626{salt: bytes32(0)}(asset, euler, IEulerEToken(eTokenAddress), rewardsDistribution, address(this));
+
+        emit CreateERC4626(asset, vault);
+    }
+
+    /// @inheritdoc ERC4626Factory
+    function computeERC4626Address(ERC20 asset) external view virtual override returns (ERC4626 vault) {
+        vault = ERC4626(
+            _computeCreate2Address(
+                keccak256(
+                    abi.encodePacked(
+                        // Deployment bytecode:
+                        type(StakeableEulerERC4626).creationCode,
+                        // Constructor arguments:
+                        abi.encode(asset, euler, IEulerEToken(markets.underlyingToEToken(address(asset))))
+                    )
+                )
+            )
+        );
+    }
```

Comment:
aren't we missing some constructor arguments here in order to compute correctly the address?

Reply:
I didn't consider it necessary since those parameters wouldn't change, they are fixed once the factory was deployed. But I think it makes sense to add them, for consistency at least